### PR TITLE
int-log-compiler-common.sh: Remove dependency on which

### PR DESCRIPTION
**Pull request by salahcoronya**
See https://lwn.net/Articles/874049/

### CODE REVIEW

**Comment by AndreasFuchsTPM** [comment]
Why use `type -P` ?
The lwn article mentions `command -v`.

**Comment by thesamesam** [comment]
If the file relies on bash already, `type -P` is mildly superior as it will ignore functions defined in the user's environment. I haven't checked if that's the case here.

**Comment by AndreasFuchsTPM** [comment]
Given the comment in https://github.com/tpm2-software/tpm2-tss/pull/2830 I'd prefer `command -v` to have as few bash-specifics as possible in favor of posix.
Could you update the PR for me to merge ?

**Comment by eli-schwartz** [comment]
> Given the comment in #2830 I'd prefer `command -v` to have as few bash-specifics as possible in favor of posix. Could you update the PR for me to merge ?

I'm not attached to the use of type -P or bash, but just as a devil's advocate I would like to point out that you can have a policy that your tests use bash, and that's a project choice.

configure.ac *has* to be written in POSIX sh because the autotools don't let you choose the script interpreter shebang for your ./configure script -- it is always /bin/sh, never bash.

File: script/int-log-compiler-common.sh [hunk 1]
```diff
@@ -55,12 +55,12 @@ sanity_test ()
         exit 1
     fi
 
-    if [ -z "$(which ${sock_tool})" ]; then
+    if [ -z "$(command -v ${sock_tool})" ]; then
```
Comment (eli-schwartz) [inline script/int-log-compiler-common.sh:58]: You can just check `if ! command -v XXX >/dev/null; then`

**Review by AndreasFuchsTPM — APPROVED** [review]

**Pull request merged**